doc/source/user_guide/fields/field_data.rst

```diff
@@ -147,98 +161,37 @@ First, create a transaction object for field data.
   >>> transaction = solver.fields.field_data.new_transaction()
   # This creates a new transaction object for batching multiple requests.
 
-Then combine requests for multiple fields using ``add_<items>_request`` methods in a single transaction:
-
-- ``add_surfaces_request`` adds a surfaces request.
-- ``add_scalar_fields_request`` adds a scalar fields request.
-- ``add_vector_fields_request`` adds a vector fields request.
-- ``add_pathlines_fields_request`` adds a pathlines fields request.
+Then combine requests for multiple fields using ``add_requests`` methods in a single transaction
+and call the ``get_response`` method to execute the transaction and retrieve the data.
 
 .. code-block:: python
 
-  >>> transaction.add_surfaces_request(
-  >>>     surfaces=[1], data_types=[SurfaceDataType.Vertices, SurfaceDataType.FacesCentroid]
-  >>> )
-  # Adds a request for surface data such as vertices and centroids.
-  >>> transaction.add_scalar_fields_request(
-  >>>     surfaces=[1, 2], field_name="pressure", node_value=True, boundary_value=True
-  >>> )
-  # Adds a request for scalar field data like pressure.
-  >>> transaction.add_vector_fields_request(
-  >>>     surfaces=[1, 2], field_name="velocity"
-  >>> )
-  # Adds a request for vector field data like velocity.
-
-You can call the ``get_fields`` method to execute the transaction and retrieve the data:
+  >>> su1 = SurfaceFieldDataRequest(surfaces=[1], data_types=[SurfaceDataType.Vertices, SurfaceDataType.FacesCentroid])
+  >>> sc1 = ScalarFieldDataRequest(surfaces=[1, 2], field_name="pressure", node_value=True, boundary_value=True)
+  >>> vc1 = VectorFieldDataRequest(surfaces=[1, 2], field_name="velocity")
 
-.. code-block:: python
+  >>> payload_data = transaction.add_requests(su1, sc1, vc1).get_response()
 
-  >>> payload_data = transaction.get_fields()
-  # Executes all requests and returns the combined field data.
+You can use the ``get_field_data`` method on ``payload_data`` and retrieve the data in the same manner
+like for getting field data. You can either reuse the same request object or form a new one by copying
+from the existing one of similar type:
 
-``payload_data`` is a dictionary containing the requested fields as a numpy array in the following order:
+.. code-block:: python
 
-``tag -> surface_id [int] -> field_name [str] -> field_data[np.array]``
+  >>> scalar_field_data = payload_data.get_field_data(sc1)
+  >>> scalar_field_data.keys()
+  dict_keys([1, 2])
+  >>> sc1 = sc1._replace(surfaces=[1])
+  >>> scalar_field_data = payload_data.get_field_data(sc1)
+  >>> scalar_field_data.keys()
+  dict_keys([1])
 
 .. note::
-   ``get_fields`` call also clears all requests, so that subsequent calls to this method
+   ``get_response`` call also clears all requests, so that subsequent calls to this method
```

Comment:
@prmukherj 
- This is an incomplete sentence.
- Why does `get_response()` need to clear the transaction? What if I want to reuse it to get a new snapshot of the data? For instance, if a user writes a loop to build a series of snapshots, are we proposing that the transaction must be rebuilt at each iteration?